pkg/cloud/services/networking/network.go

```diff
@@ -147,6 +150,33 @@ func (s *Service) ReconcileNetwork(openStackCluster *infrav1.OpenStackCluster, c
 	return nil
 }
 
+func (s *Service) DeletePorts(openStackCluster *infrav1.OpenStackCluster) error {
+	networkID := openStackCluster.Spec.Network.ID
+	portList, err := s.client.ListPort(ports.ListOpts{
+		NetworkID:   networkID,
+		DeviceOwner: "",
+	})
+	if err != nil {
+		if capoerrors.IsNotFound(err) {
+			return nil
+		}
+		return fmt.Errorf("searching for no owner port error: %v", err)
+	}
+
+	for _, port := range portList {
+		if strings.Contains(port.Name, openStackCluster.Name) && port.DeviceOwner == "" {
+			err := s.DeletePort(openStackCluster, port.ID)
+			if capoerrors.IsNotFound(err) {
+				continue
+			} else {
+				return fmt.Errorf("delete port %s of network %q failed : %v", port.ID, networkID, err)
+			}
```

Comment:
If the error is nil, we would also run into the `else` case.

Maybe rewrite it to something like this:
```go
if err := s.DeletePort(openStackCluster, port.ID); err != nil {
    if capoerrors.IsNotFound(err) {
        continue
    }
    return fmt.Errorf("delete port %s of network %q failed : %v", port.ID, networkID, err)
}
```

Reply:
Thanks for your suggestions,I have fixed them,please have a re-check